src/Pwa/Controller/PageController.php

```diff
@@ -12,153 +12,42 @@
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Routing\Annotation\Route;
-use OpenApi\Annotations as OA;
 
 /**
  * @Route(defaults={"_routeScope"={"store-api"}})
  */
 class PageController extends AbstractController
 {
-    /**
-     * Placeholder, because these routes names may change during implementation
-     *
-     * string
-     */
     const PRODUCT_PAGE_ROUTE = 'frontend.detail.page';
 
     const NAVIGATION_PAGE_ROUTE = 'frontend.navigation.page';
 
     const LANDING_PAGE_ROUTE = 'frontend.landing.page';
 
-    /**
-     * @var PageLoaderContextBuilderInterface
-     */
-    private $pageLoaderContextBuilder;
+    private array $pageLoaders;
 
     /**
-     * @var PageLoaderInterface[]
+     * @param PageLoaderContextBuilderInterface $pageLoaderContextBuilder
+     * @param iterable $pageLoaderIterators
      */
-    private $pageLoaders;
-
-    public function __construct(PageLoaderContextBuilderInterface $pageLoaderContextBuilder, iterable $pageLoaders)
-    {
-        $this->pageLoaderContextBuilder = $pageLoaderContextBuilder;
+    public function __construct(
+        private readonly PageLoaderContextBuilderInterface $pageLoaderContextBuilder,
+        private readonly iterable $pageLoaderIterators
+    ) {
+        $pageLoaderIteratorsAsArray = \iterator_to_array($this->pageLoaderIterators);
 
         /** @var PageLoaderInterface $pageLoader */
-        foreach ($pageLoaders as $pageLoader) {
+        foreach ($pageLoaderIteratorsAsArray as $pageLoader) {
             $this->pageLoaders[$pageLoader->getResourceType()] = $pageLoader;
         }
     }
 
     /**
      * @Route("/store-api/pwa/page", name="store-api.pwa.cms-page-resolve", methods={"POST"})
      *
-     * @OA\Post(
```

Comment:
The API schema is important for client generation and documentation purposes. This should be added as a separate file (including descriptions, parameters etc.) - as described in [this documentation](https://developer.shopware.com/docs/guides/plugins/plugins/framework/store-api/add-store-api-route.html#add-route-to-swagger).

Reply:
I've added it [here](https://github.com/shopware/SwagShopwarePwa/pull/113/commits/feae3cc5e0f57785c6463e110ba5ab747dee884f)